### PR TITLE
[ODPR-1609] - R1.2 Send Platform Operators & Reporting Notifications emails

### PR DESCRIPTION
**Pull request by jassalrichy**

### CODE REVIEW

File: app/models/email/requests/SendEmailRequest.scala [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  def apply(email: String,
+            name: String,
+            businessName: String,
+            platformOperatorId: String): UpdatedPlatformOperatorRequest = UpdatedPlatformOperatorRequest(
+    to = List(email),
+    templateId = PlatformOperatorUpdatedTemplateId,
+    parameters = Map(
+      "userPrimaryContactName" -> name,
+      "poBusinessName" -> businessName,
+      "poId" -> platformOperatorId)
```
Comment (georgievh) [inline app/models/email/requests/SendEmailRequest.scala:180]: This field is not needed for the template

File: app/models/email/requests/SendEmailRequest.scala [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+  def apply(email: String,
+            name: String,
+            businessName: String,
+            platformOperatorId: String): UpdatedAsPlatformOperatorRequest = UpdatedAsPlatformOperatorRequest(
+    to = List(email),
+    templateId = PlatformOperatorUpdatedTemplateId,
+    parameters = Map(
+      "poPrimaryContactName" -> name,
+      "poBusinessName" -> businessName,
+      "poId" -> platformOperatorId)
```
Comment (georgievh) [inline app/models/email/requests/SendEmailRequest.scala:209]: This field is not needed.

**Comment by georgievh** [comment]
As discussed we need to do refactoring and improve code coverage which we will do as part of a different ticket.
This still requires a removal of a couple of fields that are not needed by the templates. Those are related to the update PO templates.

**Review by georgievh — CHANGES_REQUESTED** [review]

File: app/connectors/EmailConnector.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, StringContextOps}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+@Singleton
+class EmailConnector @Inject()(appConfig: FrontendAppConfig, httpClient: HttpClientV2)
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/connectors/EmailConnector.scala:32]: Added.

File: app/controllers/add/CheckYourAnswersController.scala [hunk 4]
```diff
@@ -84,6 +88,8 @@ class CheckYourAnswersController @Inject()(
   def onSubmit(): Action[AnyContent] = (identify andThen getData(None) andThen requireData).async {
     implicit request =>
 
+      implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/controllers/add/CheckYourAnswersController.scala]: Updated.

File: test/models/email/requests/AddedPlatformOperatorRequestSpec.scala [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          "poId" -> "some-po-id")
+      )
+    }
+  }
+
+  ".build(...)" - {
+    "must return correct AddedPlatformOperatorRequest" in {
+      val answers = anEmptyUserAnswer.copy(operatorId = Some("some-operator-id"))
+        .set(PrimaryContactEmailPage, "email@example.com").success.value
+        .set(PrimaryContactNamePage, "some-name").success.value
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply test/models/email/requests/AddedPlatformOperatorRequestSpec.scala:52]: Removed.

File: app/models/email/requests/SendEmailRequest.scala (hunk 2 above)
Comment (jassalrichy) [reply app/models/email/requests/SendEmailRequest.scala]: Removed.

**Comment by georgievh** [comment]
Approving and a ticket will be created to address test coverage and comments.

**Review by georgievh — APPROVED** [review]

**Pull request merged**